src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs

```diff
@@ -111,13 +112,28 @@ protected bool ValidateNameConflictsInScope(Symbol symbol, Location location, st
                     break;
                 }
 
+                // symbols inside local function shadow names in scopes outside local function
+                var method = (binder as InMethodBinder)?.ContainingMemberOrLambda as MethodSymbol;
+                if (method?.MethodKind == MethodKind.LocalFunction)
+                {
+                    switch (symbol?.Kind)
+                    {
+                        case SymbolKind.Local:
+                        case SymbolKind.Parameter:
+                        case SymbolKind.TypeParameter:
+                            Debug.Assert(!error); // If the assert fails, add a corresponding test.
+                            goto End;
+                    }
+                }
```

Comment:
Do we have a test where the name of the local function is in conflict with a variable in scope?
```C#
   public void M(int local, int local2) 
    {
        // both should be errors
        void local() { }
        static void local2() { }    
    }
```